Formula/dsvpn.rb

```diff
@@ -0,0 +1,57 @@
+class Dsvpn < Formula
+  desc "Dead Simple VPN"
+  homepage "https://github.com/jedisct1/dsvpn"
+  url "https://github.com/jedisct1/dsvpn.git",
+      :tag      => "0.1.4",
+      :revision => "3b99d2ef6c02b2ef68b5784bec8adfdd55b29b1a"
+  head "https://github.com/jedisct1/dsvpn.git"
+
+  def install
+    system "make"
+    bin.install "dsvpn"
+  end
+
+  test do
+    expected = <<~EOF
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
> Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

I was anticipating this contention. I cannot do anything better than a version check here: this is not an input/output type program like a compiler or a translator. Since the functionality is VPN and that is all it does the best I can do is test client mode VPN. The problem however is that it requires sudo to run and may permanently modify system settings if it executes in a stale state. Also since I need to have a Linux server it is hard to test the VPN in a single test execution and offline.

Reply:
Nevermind I think I can actually test failing to connect with a correct error message.

Reply:
Okay it is updated now to test connecting to 127.0.0.1 port 0 with a key from /dev/zero. This should be a repeatable predictable failure case regardless of what Mac system it is run on.